awx/main/tasks/jobs.py

```diff
@@ -802,6 +640,9 @@ class RunJob(BaseTask):
     event_model = JobEvent
     event_data_key = 'job_id'
 
+    def __init__(self):
+        self.runner_callback.event_data_key = self.event_data_key
```

Comment:
I expect this will break inventory update runs, because you didn't defer to the `super` method. The proper way to override `__init__` in a subclass is that you call the super's method (like `super(self, RunJob).__init__(*args, **kwargs)`).

But if you take my advice from a prior comment, you shouldn't need this method at all.